src/components/PredefinedLinks.js

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Link from '../components/Link';
+
+// TODO: Is it possible to rewrite this without usage of return but keeping propTypes check?
```

Comment:
I've looked into this in the past, you can do it with an anonymous class:

```
const LinkCreator = path =>
  class extends React.PureComponent {
    static propTypes = {
      children: PropTypes.node.isRequired,
    };
    render() {
      return <Link {...props} path={path}>{children}</Link>;
    }
  };
```
… but that looks even worse than what we started with :)

I did a bit more research, it *looks* like `Object.assign` can be used in this case:

```
const LinkCreator = path =>
  Object.assign(
    ({ children, ...props }) => <Link {...props} path={path}>{children}</Link>,
    { propTypes: {
      children: PropTypes.node.isRequired
    }});
```

With this as a basis, we can make a little helper function to clean up the code:

```
const withProps = (component, propTypes, defaultProps) =>
  Object.assign(component, { propTypes, defaultProps });

const LinkCreator = path => withProps(
  ({ children, ...props }) => <Link {...props} path={path}>{children}</Link>,
  {
    children: PropTypes.node.isRequired,
  });
```

This `withProps` is kinda cute, maybe we should have it in `src/utils.js`.